curly/chunker_test.go

```diff
@@ -0,0 +1,109 @@
+package main
+
+import (
+	"bytes"
+	"github.com/google/go-cmp/cmp"
+	"github.com/stretchr/testify/assert"
+	"testing"
+)
+
+var _ Chunker = &testChunked{}
+
+type testChunked struct {
+	multiBuffer []*bytes.Buffer
+	buf         *bytes.Buffer
```

Comment:
nit: imho `buf` is not needed you can always access it as multiBuffer[len(multiBuffer)-1]